python/ray/_private/job_manager/job_manager.py

```diff
@@ -207,64 +246,88 @@ def _get_current_node_resource_key(self) -> str:
             raise ValueError(
                 "Cannot found the node dictionary for current node.")
 
-    def submit_job(
-            self,
-            entrypoint: str,
-            runtime_env: Optional[Dict[str, Any]] = None,
-            metadata: Optional[Dict[str, str]] = None,
-    ) -> str:
+    def submit_job(self,
+                   entrypoint: str,
+                   runtime_env: Optional[Dict[str, Any]] = None,
+                   metadata: Optional[Dict[str, str]] = None,
+                   _start_signal_actor: Optional[SignalActor] = None) -> str:
         """
-        1) Create new detached actor with same runtime_env as job spec
-        2) Get task / actor level runtime_env as env var and pass into
+        Job execution happens asynchronously.
+
+        1) Generate a new unique id for this job submission, expected to be
+            executed in complete isolation.
+        2) Create new detached actor with same runtime_env as job spec
+        3) Get task / actor level runtime_env as env var and pass into
             subprocess
-        3) subprocess.run(entrypoint)
+        4) asyncio.create_subprocess_shell(entrypoint)
```

Comment:
This comment makes it seem like (3) and (4) are done in this function, but they happen in the actor. Maybe just specify that subprocess management is all handled by the actor?.